internal/intentapi/apiserver_proxy.go

```diff
@@ -0,0 +1,98 @@
+package intentapi
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/setting"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubectl/pkg/proxy"
+)
+
+// ApiserverProxyConfig is the config for ApiserverProxy.
+type ApiserverProxyConfig struct {
+	RestCfg          *rest.Config
+	KeepaliveTimeout time.Duration
+}
+
+// NewApiserverProxyConfig parses and returns a new ApiserverProxyConfig.
+func NewApiserverProxyConfig(cfg *setting.Cfg) (ApiserverProxyConfig, error) {
+	sec := cfg.Raw.Section("intentapi.proxy")
+	configPath := sec.Key("kubeconfig_path").MustString("")
+
+	if configPath == "" {
+		return ApiserverProxyConfig{}, errors.New("kubeconfig path cannot be empty when using Intent API")
+	}
+
+	configPath = filepath.Clean(configPath)
+
+	if _, err := os.Stat(configPath); os.IsNotExist(err) {
+		return ApiserverProxyConfig{}, fmt.Errorf("cannot find kubeconfig file at '%s'", configPath)
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", configPath)
+	if err != nil {
+		return ApiserverProxyConfig{}, err
+	}
+
+	return ApiserverProxyConfig{
+		RestCfg:          config,
+		KeepaliveTimeout: sec.Key("keepalive_timeout").MustDuration(1 * time.Minute),
+	}, nil
+}
+
+// ApiserverProxy is a proxy for kube-apiserver.
+type ApiserverProxy struct {
+	config  ApiserverProxyConfig
+	handler http.Handler
+	logger  log.Logger
+}
+
+// ProvideApiserverProxy provides a new ApiserverProxy with given configuration.
+func ProvideApiserverProxy(cfg *setting.Cfg, features featuremgmt.FeatureToggles) (*ApiserverProxy, error) {
```

Comment:
How are we relating this to the bridge package we've already created? Are we leaving that to a different PR?

Reply:
Since current implementation is just a dumb proxy, based on `kubectl proxy` machinery, I think it's safe to leave the bridge out, until we find some use for it.

I'm currently not 100% sure what kind of logic we'll end up putting into the backend later on, so I've decided to go the simple route for now.

Reply:
I think that makes sense, but to be clear: we're not suggesting that the bridge and proxy are lying on a separate underlying API server, right?

meaning that a future PR would end up adding new parameters to the bridge provider that consume the same settings which of an introduced to it

Reply:
Ah yes, that's a good point. Yeah, that's the idea here. Now that you've said it, we could move the config part to the bridge and import the bridge here. And maybe even move `kube-proxy` there?

Reply:
Okay, I've moved config initialisation and exporting to k8s bridge, so that the config can be re-used between multiple other components. I've left `kube-proxy` in `ApiserverProxy` for now, I think that makes more sense here.